iothub/device/src/InternalClient.cs

```diff
@@ -322,6 +323,24 @@ public async Task<Message> ReceiveAsync()
                     return await ReceiveAsync(cts.Token).ConfigureAwait(false);
                 }
             }
+            catch (ObjectDisposedException)
+            {
+                bool isObjectDisposedExceptionDisabled = false;
+#if !NET451
+                AppContext.TryGetSwitch(AppContextConstants.DisableObjectDisposedExceptionForReceiveAsync, out isObjectDisposedExceptionDisabled);
```

Comment:
What if TryGetSwitch returns false? Please either add a check or comment explaining why it's ignored.